modules/transport-netty4/src/main/java/org/elasticsearch/transport/netty4/NettyAllocator.java

```diff
@@ -170,11 +169,8 @@ public int pageSize() {
         };
     }
 
-    @SuppressForbidden(
-        reason = "TODO Deprecate any lenient usage of Boolean#parseBoolean https://github.com/elastic/elasticsearch/issues/128993"
-    )
     private static boolean useG1GC() {
-        return Boolean.parseBoolean(JvmInfo.jvmInfo().useG1GC());
+        return org.elasticsearch.common.util.Booleans.parseBoolean(JvmInfo.jvmInfo().useG1GC());
```

Comment:
https://github.com/elastic/elasticsearch/blob/ffb7782324f989751a647ce4a3198b7fb9630f1f/server/src/main/java/org/elasticsearch/monitor/jvm/JvmInfo.java#L84

Based on above, we probably should use `Booleans.parseBooleanLenient`